[Docs] [istio mtls] Add warning on sidecar OOM for mTLS (#53385)
ray-project/ray: doc/source/cluster/kubernetes/k8s-ecosystem/istio.md

:::

:::{warning}

> The default Ray worker port range, from 10002 to 19999, is too large to specify in the service manifest and can cause memory issues in Kubernetes. Set a smaller `max-worker-port` to work with Istio. Note that by default these ports will be cached in every sidecar in the service mesh, which could lead to sidecar OOMs if too many headless services are created. 

Reviewer: > We just had a production outage at Roblox due to a large number of headless services created for Ray overwhelming the service mesh sidecars and causing ingress gateways to fail.

Just out of curiosity, how many ports did you set?

Reviewer: Have you installed https://docs.ray.io/en/latest/ray-contribute/docs.html#how-do-you-run-vale?

Author: We set 200 ports to still allow 200 workers per node in the Ray cluster. However, people queued up hundreds of jobs and caused every infra pod with sidecar to OOM

> The default Ray worker port range, from 10002 to 19999, is too large to specify in the service manifest and can cause memory issues in Kubernetes. Set a smaller `max-worker-port` to work with Istio. Note that by default these ports will be cached in every sidecar in the service mesh, which could lead to sidecar OOMs if too many headless services are created. 

Reviewer 2: ```suggestion
The default Ray worker port range, from 10002 to 19999, is too large to specify in the service manifest and can cause memory issues in Kubernetes. Set a smaller `max-worker-port` to work with Istio. Note that by default every sidecar in the service mesh caches these ports, which could lead to sidecar OOMs if you create too many headless services. 
```

Reviewer 2: Trying to clarify what is being done by whom/what. Passive voice is inherently ambiguous. Please correct if the subjects are inaccurate.

Author: Thanks!
